lib/src/Uncertainty/Algorithm/Simulation/ProbabilitySimulation.cxx

```diff
@@ -0,0 +1,123 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief ProbabilitySimulation is a generic view of simulation methods for computing
+ * probabilities and related quantities by sampling and estimation
+ *
+ *  Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/ProbabilitySimulation.hxx"
+#include "openturns/Log.hxx"
+#include "openturns/Curve.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/ResourceMap.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/*
+ * @class ProbabilitySimulation
+ */
+
+CLASSNAMEINIT(ProbabilitySimulation);
+
+static const Factory<ProbabilitySimulation> Factory_ProbabilitySimulation;
+
+
+/* Constructor with parameters */
+ProbabilitySimulation::ProbabilitySimulation(const Event & event,
+                                             const WeightedExperiment & experiment,
+                                             const Bool verbose,
+                                             const HistoryStrategy & convergenceStrategy)
+  : Simulation(event, verbose, convergenceStrategy)
+{
+  setExperiment(experiment);
+}
+
+ProbabilitySimulation::ProbabilitySimulation(const Bool verbose,
+                                             const HistoryStrategy & convergenceStrategy)
+  : Simulation(verbose, convergenceStrategy)
+{
+  // Nothing to do
+}
+
+/* Virtual constructor */
+ProbabilitySimulation * ProbabilitySimulation::clone() const
+{
+  return new ProbabilitySimulation(*this);
+}
+
+
+void ProbabilitySimulation::setExperiment(const WeightedExperiment & experiment)
+{
+  experiment_ = experiment;
+  experiment_.setDistribution(getEvent().getImplementation()->getAntecedent()->getDistribution());
```

Comment:
I don't see simulation tests with such events. In fact I don't see tests involving ProcessEvents. Do you have a script with that type of simulation ?